projects/fsastorefrontlib/src/cms-components/user-profile/user-profile.component.ts

```diff
@@ -58,13 +81,70 @@ export class UserProfileComponent implements OnInit, OnDestroy {
   assets: { [key: string]: any }[];
   assetSelected: AssetTableType;
   showAddressForm = false;
+  showProductOverview = true;
   userId: string;
   customerId: string;
 
+  selectedProducts = new BehaviorSubject(ProductOverviewCategory.ALL);
+  selectedProducts$ = this.selectedProducts.asObservable();
+
+  allProducts$: Observable<CmsComponent[]>;
+  productsCountByCategory$: Observable<{
+    [key: string]: typeof ProductOverviewCategory;
+  }>;
+  filteredProducts$: Observable<CmsComponent[]>;
+
   ngOnInit(): void {
+    this.allProducts$ = this.componentData.data$.pipe(
+      map(data => data.children.split(' ')),
+      switchMap(children =>
+        from(children).pipe(
+          mergeMap(child => this.cmsService.getComponentData(child)),
+          take(children.length),
+          toArray()
+        )
+      )
+    );
+
+    const initialProductsCountObj = Object.values(
+      ProductOverviewCategory
+    ).reduce((prev, curr, i) => {
+      if (!prev[curr]) prev[curr] = 0;
+      return prev;
+    }, {});
+
+    this.productsCountByCategory$ = this.allProducts$.pipe(
+      mergeMap(components =>
+        from(components).pipe(
+          reduce((prev, curr) => {
+            prev[curr.category]++;
+            prev['all']++;
+            return prev;
+          }, initialProductsCountObj),
+          take(components.length)
+        )
+      ),
+      shareReplay()
+    );
```

Comment:
Similar comment as above. Try to move to separate method for better readability and easier maintenance.